src/Runner.php

```diff
@@ -11,15 +11,24 @@
 class Runner
 {
     protected $index;
+
+    /**
+     * @var MiddlewareStack
+     */
     protected $middleware;
 
-    public function run($middleware, ServerRequestInterface $request, ResponseInterface $response)
+    public function run(MiddlewareStack $middleware, ServerRequestInterface $request, ResponseInterface $response)
```

Comment:
I'm no longer sure it is a good idea to type hint on concrete implementations. It makes swapping implementations out harder than no typehint does.